viewing_party/party.py

```diff
@@ -1,23 +1,145 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    if not (title and genre and rating):
+        return None
+    else:
+        movie = { "title" : title,
+                    "genre" : genre,
+                    "rating" : rating}
+        return movie
+    
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+
+    return user_data    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    if not user_data["watched"]:
+        return 0.0
+    else:
+        total_rating = sum(movie["rating"] for movie in user_data["watched"])
+        return total_rating / len(user_data["watched"])

```

Comment:
While there's no official ruling one way or the other, I like to remove the else statement entirely if it isn't necessary. This is especially true when the if portion of the conditional is a guard clause.